src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -1,33 +1,98 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.api.accounts.Kind;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import uk.gov.companieshouse.api.accounts.LinkType;
 import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
 import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
 import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
 
 @Service
 public class CompanyAccountServiceImpl extends
         AbstractServiceImpl<CompanyAccount, CompanyAccountEntity> implements CompanyAccountService {
 
+    @Autowired
+    private TransactionManager transactionManager;
+
     @Autowired
     public CompanyAccountServiceImpl(
             @Qualifier("companyAccountRepository") MongoRepository<CompanyAccountEntity, String> mongoRepository,
             @Qualifier("companyAccountTransformer") GenericTransformer<CompanyAccount, CompanyAccountEntity> transformer) {
         super(mongoRepository, transformer);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public CompanyAccount createCompanyAccount(CompanyAccount companyAccount) {
+        HttpServletRequest request = getRequestFromContext();
+        ResponseEntity<Transaction> transaction = getTransactionFromSession(request);
+
+        addKind(companyAccount);
+        addEtag(companyAccount);
+
+        CompanyAccountEntity companyAccountEntity = genericTransformer.transform(companyAccount);
+
+        addEntityID(companyAccountEntity);
+        addEntityLinks(companyAccountEntity, transaction);
+
+        mongoRepository.insert(companyAccountEntity);
+
+        transactionManager.updateTransaction(transaction.getBody().getId(), request.getHeader("X-Request-Id"), getCompanyAccountSelfLink(companyAccountEntity));
+
+        return genericTransformer.transform(companyAccountEntity);
+    }
+
+    public void addEntityLinks(CompanyAccountEntity entity, ResponseEntity<Transaction> transaction) {
+        Map<String, String> map = new HashMap<>();
+        String selfLink = createCompanyAccountSelfLink(entity, transaction);
+        map.put(LinkType.SELF.getLink(), selfLink);
+        entity.getData().setLinks(map);
+    }
+
+    private String createCompanyAccountSelfLink(CompanyAccountEntity entity, ResponseEntity<Transaction> transaction) {
+        return getTransactionSelfLink(transaction) + "/company-accounts/" + entity.getId() + "";
```

Comment:
An empty string is being added to the link?

Reply:
Yeh good spot. Fixed in [84aa56c](https://github.com/companieshouse/company-accounts.api.ch.gov.uk/pull/18/commits/84aa56c9f1d5acd2ce6b9a79332a164d195a364d)